File: backend/app/schemas/cliente.py
"""Pydantic schemas for Cliente entity."""

from datetime import datetime
from typing import Optional

from pydantic import BaseModel, ConfigDict, EmailStr, Field


class ClienteBase(BaseModel):
    nombre: str = Field(min_length=1, max_length=120)
    email: Optional[EmailStr] = None
    telefono: Optional[str] = Field(default=None, max_length=30)
    direccion: Optional[str] = Field(default=None, max_length=255)
    barrio: Optional[str] = Field(default=None, max_length=120)


class ClienteCreate(ClienteBase):
    pass


class ClienteUpdate(BaseModel):
    nombre: Optional[str] = Field(default=None, min_length=1, max_length=120)
    email: Optional[EmailStr] = None
    telefono: Optional[str] = Field(default=None, max_length=30)
    direccion: Optional[str] = Field(default=None, max_length=255)
    barrio: Optional[str] = Field(default=None, max_length=120)


class ClienteResponse(ClienteBase):
    model_config = ConfigDict(from_attributes=True)

    id: int
    negocio_id: int
    created_at: datetime

File: frontend/src/types/cliente.ts
export interface Cliente {
  id: number;
  nombre: string;
  email?: string | null;
  telefono?: string | null;
  direccion?: string | null;
  barrio?: string | null;
  created_at?: string;
}

export interface ClientePayload {
  nombre: string;
  email?: string;
  telefono?: string;
  direccion?: string;
  barrio?: string;
}
